### ISSUE TITLE
[Training] [M3.5] Retrain student model on hydrogen-complete dataset with angular loss

### ISSUE THREAD

**Issue by atfrank** [atfrank/MLFF-Distiller#21]
## Problem Statement

Previous training (100 epochs) was performed on dataset lacking hydrogen atoms (0% H). This resulted in:
- Force RMSE: 0.87 eV/Å (8.7x above 0.1 eV/Å target)
- Angular error: 16.67° mean (only 45.5% atoms <5°)
- Poor oxygen predictions (80% worse than carbon)

With hydrogen-complete dataset and angular loss component, we expect 3-5x improvement in force accuracy.

## Background

**Critical bug fixed**: Dataset now includes explicit hydrogen atoms (~50% of total atoms)
**Enhancement added**: Angular loss component in `src/mlff_distiller/training/losses.py`

New loss formula:
```
total_loss = energy_weight * E_MSE + 
             force_weight * F_MSE +
             angular_weight * (1 - cos(θ))
```

## Tasks

- [ ] **Wait for Issue #20 completion** (BLOCKER)
  - Dependency: Hydrogen-complete dataset at `data/merged_dataset_with_H/dataset.h5`
  - Verify dataset ready with `scripts/verify_hydrogen_fix.py`
  
- [ ] **Configure training hyperparameters**
  - Review current config: `configs/student_training_config.yaml`
  - Set angular_weight: 10.0 (initial value)
  - Confirm: epochs=100, batch_size=8
  
- [ ] **Run training with angular loss**
  ```bash
  python scripts/train_student.py \
    --data data/merged_dataset_with_H/dataset.h5 \
    --epochs 100 \
    --batch-size 8 \
    --energy-weight 1.0 \
    --force-weight 100.0 \
    --angular-weight 10.0 \
    --run-name "h_complete_angular_loss"
  ```
  - Expected duration: ~2 hours
  
- [ ] **Monitor training metrics**
  - Track: Energy MSE, Force MSE, Angular error
  - Target: Angular error <10° by epoch 50, Force RMSE <0.25 eV/Å by epoch 100
  
- [ ] **Compare with baseline training**
  - Load baseline: `checkpoints/checkpoint_epoch_36.pt`
  - Create comparison plots
  
- [ ] **Generate training report**
  - Create: `docs/H_COMPLETE_TRAINING_RESULTS.md`
  - Include: metrics, curves, comparison, per-element analysis

## Acceptance Criteria

- [ ] Force RMSE < 0.25 eV/Å (3.5x better than baseline)
- [ ] Angular error < 10° mean
- [ ] Energy MAE < 5 meV/atom
- [ ] O/C performance gap <20% (vs baseline 80%)
- [ ] Smooth convergence with all loss components decreasing

## Dependencies

**Depends On**: Issue #20 (hydrogen dataset - BLOCKER)
**Blocks**: Issue #21 (validation analysis)

**Expected Improvements**:
- Force RMSE: 0.87 → <0.25 eV/Å (3.5x)
- Angular Error: 16.67° → <10° (40% reduction)

## Estimated Effort

~5 hours (2h GPU training + 3h analysis/docs)
